src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -713,7 +713,9 @@ async Task<bool> NextResultSchemaOnly(bool async, bool isConsuming = false, Canc
                         break;
                     case BackendMessageCode.RowDescription:
                         // We have a resultset
-                        RowDescription = _statements[StatementIndex].Description = (RowDescriptionMessage)msg;
+                        RowDescription = _statements[StatementIndex].Description = preparedStatement == null
```

Comment:
nit: copy over the comment from the non-SchemaOnly codepath to explain why we're cloning